app/src/main/java/eu/kanade/presentation/updates/components/calendar/Calendar.kt

```diff
@@ -0,0 +1,163 @@
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import kotlinx.collections.immutable.ImmutableMap
+import kotlinx.collections.immutable.toImmutableList
+import java.time.DayOfWeek
+import java.time.LocalDate
+import java.time.Month
+import java.time.format.TextStyle
+import java.util.Locale
+import kotlin.math.ceil
+
+private val CalenderPadding = 8.dp
+private val FontSize = 16.sp
+private const val ExtendedScale = 0.31f
+private const val MediumScale = 0.60f
+private const val HeightMultiplier = 68
+private const val DaysOfWeek = 7
+
+@Composable
+fun Calendar(
+    events: ImmutableMap<LocalDate, Int>,
+    screenWidth: Dp,
+    modifier: Modifier = Modifier,
+    isTabletUi: Boolean = false,
+    labelFormat: (DayOfWeek) -> String = {
+        it.getDisplayName(
+            TextStyle.SHORT,
+            Locale.getDefault(),
+        )
+    },
+    onClickDay: (day: LocalDate) -> Unit = {},
+) {
+    val today = LocalDate.now()
+    val weekValue = remember { DayOfWeek.entries.toTypedArray() }
+    val displayedMonth = remember { mutableStateOf(today.month) }
+    val displayedYear = remember { mutableIntStateOf(today.year) }
+    val currentMonth = displayedMonth.value
+    val currentYear = displayedYear.intValue
+
+    val widthModifier = when {
+        isTabletUi -> 1.0f
+        screenWidth > 840.dp -> ExtendedScale
+        screenWidth > 600.dp -> MediumScale
+        else -> 1.0f
+    }
+
+    Column(
+        modifier = modifier
+            .wrapContentHeight()
+            .fillMaxWidth()
+            .padding(all = CalenderPadding),
+        horizontalAlignment = Alignment.CenterHorizontally,
+    ) {
+        CalenderHeader(
+            month = currentMonth,
+            year = currentYear,
+            onPreviousClick = {
+                displayedYear.intValue -= if (currentMonth == Month.JANUARY) 1 else 0
+                displayedMonth.value -= 1
+            },
+            onNextClick = {
+                displayedYear.intValue += if (currentMonth == Month.DECEMBER) 1 else 0
+                displayedMonth.value += 1
+            },
+        )
+        Spacer(modifier = Modifier.padding(vertical = 4.dp))
+        Row(
+            modifier = Modifier.padding(horizontal = 8.dp),
+        ) {
+            CalendarGrid(
+                weekValue = weekValue,
+                labelFormat = labelFormat,
+                currentMonth = currentMonth,
+                currentYear = currentYear,
+                isTabletUi = isTabletUi,
+                events = events,
+                widthModifier = widthModifier,
+                onClickDay = onClickDay,
+                modifier = Modifier.padding(horizontal = 8.dp),
+            )
+        }
+    }
+}
+
+@Composable
+private fun CalendarGrid(
+    weekValue: Array<DayOfWeek>,
+    labelFormat: (DayOfWeek) -> String,
+    currentMonth: Month,
+    currentYear: Int,
+    isTabletUi: Boolean,
```

Comment:
yeet

Reply:
Keeping it as a property inside of calendar so calendar children don't need to access global tablet state